### ISSUE TITLE
Make all HTML valid and add a HTML linter

### ISSUE THREAD

**Issue by LiHaoTan** [TEAMMATES/teammates#7499]
Malformed HTMLs are supported by browsers so websites still work, however that does not mean we should write HTML that is invalid because that can cause unexpected behavior.

See regression: #7498 

There is some work required to fix this and will require further exploration as well, see f850a6b18fd0269d2ce510c818db975009c85b87.

A possible linter that can be used would be: https://github.com/validator/validator

**Comment by damithc** [comment]
Perhaps the linter can also run over the html files used for testing? This is because some of the URLs are generated dynamically and may not be visible in jsp/jstl files

**Comment by LiHaoTan** [comment]
@damithc yes that was what I was thinking as well, maybe the linter only runs after all the tests

**Comment by wkurniawan07** [comment]
I tried examining HTML linting some time back, but ultimately decided to put it on hiatus because of shift in focus. Some of thing I noticed back then (may or may not still apply now):
- Most/all HTML linters won't be effective in linting JSP such when encountering `<c:if>...</c:if>`, `${...}` tags.
- HTML files used for testing are not worth linting because they have been beautified for God mode regeneration.
- Linting is most effective when examining page source verbatim because it can catch the errors effectively (styles, malforms, etc.).

The linters I tried back then: [HTMLHint](https://github.com/yaniswang/HTMLHint), [htmllint](https://github.com/htmllint/htmllint).
